src/index.js

```diff
@@ -1,567 +1,363 @@
- * deriving.
- * @return {Promise => Object} a wallet derived according to BIP32 from the
- *  SHA-512 hash of the seed+password.
+ * Derive a BIP32 master node -- supplemented with a corresponding Ethereum
+ * address -- from a seed.
+ *
+ * @param  {String}  seed a BIP39 mnemonic
+ * @param  {String}  password an optional password to use when generating the
+ *   BIP39 seed
+ * @return {Object} a BIP32 node
  */
-const walletFromSeed = async (seed, password) => {
-  // we hash the seed with SHA-512 before doing BIP32 wallet generation,
-  // because BIP32 doesn't support seeds of bit-lengths < 128 or > 512.
-  const seedHash = await hash(seed, defaultTo(password, ''));
-  const { publicKey, privateKey, chainCode } = bip32.fromSeed(Buffer.from(seedHash));
-  return {
-    public: buf2hex(publicKey),
-    private: buf2hex(privateKey),
-    chain: buf2hex(chainCode),
-  };
-};
+const bip32NodeFromSeed = (mnemonic, password) => {
+  const seed = bip39.mnemonicToSeed(mnemonic, password)
+  const hd = hdkey.fromMasterSeed(seed)
+  const path = "m/44'/60'/0'/0/0"
+  const wallet = hd.derive(path)
 
+  const publicKey = buf2hex(wallet.publicKey)
+  const privateKey = buf2hex(wallet.privateKey)
+  const chain = buf2hex(wallet.chainCode)
+  const address = addressFromSecp256k1Public(publicKey)
 
-
-/**
- * Wraps nacl.lowlvel.crypto_hash
- * @param  {Uint8Array} seed
- * @return {Array}
- */
-const naclHash = seed => {
-  let newHash = []
-  nacl.lowlevel.crypto_hash(newHash, seed.reverse(), seed.length)
-  return newHash
+  return {
+    public: publicKey,
+    private: privateKey,
+    chain,
+    address
+  }
 }
 
 
 
 /**
- * Derive Urbit network keypairs from a seed. Matches ++pit:nu:crub:crypto
+ * Derive Urbit network keypairs from a seed.  Matches ++pit:nu:crub:crypto
  * @param  {Buffer} seed     seed to derive from
- * @param  {Buffer} password optional password to salt the seed before deriving
- * @return {object} urbitKeys, derived according to ++pit:nu:crub:crypto.
+ * @return {Object} resulting Urbit network keys
  */
-const urbitKeysFromSeed = (seed, password) => {
-  const h = naclHash(Buffer.concat([seed, password]));
+const urbitKeysFromSeed = seed => {
+  let h = []
+  nacl.lowlevel.crypto_hash(h, seed.reverse(), seed.length)
 
-  const c = h.slice(32);
-  const a = h.slice(0, 32);
+  const c = h.slice(32)
+  const a = h.slice(0, 32)
 
-  const crypt = nacl.sign.keyPair.fromSeed(Buffer.from(c));
-  const auth = nacl.sign.keyPair.fromSeed(Buffer.from(a));
+  const crypt = nacl.sign.keyPair.fromSeed(Buffer.from(c))
+  const auth = nacl.sign.keyPair.fromSeed(Buffer.from(a))
+
+  const crypt_pub = buf2hex(crypt.publicKey.reverse())
+  const auth_pub = buf2hex(auth.publicKey.reverse())
 
   return {
     crypt: {
       private: buf2hex(c.reverse()),
-      public: buf2hex(crypt.publicKey.reverse()),
+      public: crypt_pub,
+      address: addressFromNetworkPublic(crypt_pub)
```

Comment:
As per another comment below: no point in deriving Ethereum addresses from Urbit keys.